stdlib/public/Reflection/TypeLowering.cpp

```diff
@@ -703,21 +745,31 @@ class BitMask {
 
   template<typename IntegerType>
   void andMask(IntegerType value, unsigned byteOffset) {
-    andMask((void *)&value, sizeof(value), byteOffset);
+    if (byteOffset >= size) {
+      makeZero();
```

Comment:
This seems to imply that `andMask()` does a bitwise and of the `BitMask` with `andMask()`'s argument, treating the latter as zero extended in either direction. However, the implementation of `andMask()` below doesn't seem to match that behaviour — it doesn't zero bytes in the `BitMask` that are not covered by the `andMask()`'s argument.

Reply:
Thanks for pointing that out.  It actually looks like the uses of it below expect that the argument is one-extended in either direction.  I'll comment that expectation and fix up the implementation here to be consistent.